### PR TITLE
Fix torch.utils.checkpoint not showing up in doc

### PR DESCRIPTION
**Pull request by ssnl**
Add checkpoint to index.rst. Order torch.utils.* alphabetically.

cc @prigoyal

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -34,12 +34,13 @@ PyTorch is an optimized tensor library for deep learning using GPUs and CPUs.
    torch.distributed <distributed>
    torch.legacy <legacy>
    cuda
+   bottleneck
+   checkpoint
```
Comment (apaszke) [inline docs/source/index.rst]: Does it display correctly with torch.utils as a prefix? Take that into account when sorting them

Comment (SsnL) [reply docs/source/index.rst]: It loads the `checkpoint.rst` file I believe :)

Comment (SsnL) [reply docs/source/index.rst]: I only sorted the torch.utils.* ones (bottleneck, checkpoint, ffi, cpp_extension, model_zoo, data)

**Review by apaszke — APPROVED** [review]

**Comment by SsnL** [comment]
merging since no CI tests doc anyways

**Pull request merged**